Source/Lib/Trakt.NET/Extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace TraktNet.Extensions
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    /// <summary>Provides helper methods for IAsyncEnumerables.</summary>
+    public static class IEnumerableExtensions
+    {
+        /// <summary>Returns results from a list of tasks as they finish.</summary>
+        /// <param name="source">The list of tasks to stream results from as they finish</param>
+        /// <returns>An IAsyncEnumerable of the results from the tasks</returns>
+        public static async IAsyncEnumerable<TSource> StreamFinishedTaskResultsAsync<TSource>(this List<Task<TSource>> source)
+        {
+            if(source is null || source.Count == 0)
+                yield break;
+
+            while (source.Any())
+            {
+                Task<TSource> finishedTask = await Task.WhenAny(source).ConfigureAwait(false);
+                source.Remove(finishedTask);
+                yield return await finishedTask;
+
+            }
+        }
+
+        /// <summary>Returns results from an IEnumerable of tasks as they finish.</summary>
+        /// <param name="source">The collection of tasks to stream results from as they finish</param>
+        /// <returns>An IAsyncEnumerable of the results from the tasks</returns>
+        public static async IAsyncEnumerable<TSource> StreamFinishedTaskResultsAsync<TSource>(this IEnumerable<Task<TSource>> source)
```

Comment:
This method is only used internally?
If so, this method can be `internal` instead of `public`.

Reply:
It's only used internally currently, but it's no different than the datetime and string extensions. It allows the user of the library access to this method.

Reply:
I think this method can be removed?
It's not used, because `StreamFinishedTaskResultsAsync` is only used with `List<T>`?